### ISSUE TITLE
Plugin not refreshing remote ontologies

### ISSUE THREAD

**Issue by stuchalk** [lunisolar/LNKD.tech-Editor#45]
I am actively developing an ontology to go with my JSON-LD data.  When I update the ontology (on GitHub) I cannot find a way to refresh the plugin so it will show me the new terms I have defined.   Is there way to force a refresh?

**Comment by jak-ub** [comment]
The Git support is left to the IDE the plugin is being installed in. Basically you have a project that is local clone of your repo and that way the plugin sees the files that are checked out (and refreshed via Git support in IDE). 

If you have a two project/repo setup: 

The only way plugin sees files from outside single project is the "Global Graphs" feature of the plugin (Settings/LNKD.tech Editor/GlobalGraphs OR Settings/Othres/LNKD.tech Editor/GlobalGraphs). 
In there, you can configure  "global graphs", per project. Either by selecting Build-in graphs or adding custom graphs (second tab). In custom graphs you can add either: single files or directories.
Basically checkout locally the repo containing the ontology. Add that specific file, or folder, to the "global graphs" of the other project. Any changes to the files or directories, will be visible by the plugin. And again, Git support is left either to IDE or any GIT tool you use for that repo.

I hope this does explain the expected dynamic between plugin, IDE and GIT. Unfortunately support for remote graphs (not necessarily via GIT, but more like SPARQL endpoint) are currently only in back of my head.

Let me know if you have more questions.

**Comment by stuchalk** [comment]
Thanks for the quick answer.  I am not sure that I explained clearly enough what is going on.  The image below shows my issue.  On line 8 I import the ontology https://stuchalk.github.io/scidata/ontology/chemtwin.owl which I have updated recently to add the term 'isAtomOfElement'.  As you can see at the bottom of the image the 'cto:isAtomOfElement' is not recognized even though it is defined in the (remote) ontology (it does recognize 'cto:hasBondAtom' which was in the file when I add it to the code initially).  Are you suggesting that I need to clear caches of plug-ins though PyCharm?
![Screen Shot 2022-08-16 at 4 26 07 PM](https://user-images.githubusercontent.com/540150/184979348-a75e79bb-1a30-4abe-8ad8-6efc280f0362.jpg)

**Comment by jak-ub** [comment]
The plugin does not support RDF-XML/OWL files. 
Do I read you correctly that you added somewhere in your project (in TTL file) statements about  cto:hasBondAtom. And now you are looking for automating this process?

**Comment by stuchalk** [comment]
Sorry, I missed your reply.  I can report that I converted the OWL ontology above to turtle and it is now worse as the ontology cannot be read at all (I did this by changing the .owl in the prefix definitions to .ttl).  If I change back to .owl, it reads the owl file because it recognizes that cto:atom and cto:hasBondedAtom definitions but the cto:isAtomOfElement is not.  Once I am back in the office (week of November 7th) I would be happy to get on a call to discuss.

**Comment by jak-ub** [comment]
Closing as: 
- "Remote" ontology support was just added in version 2024.0.0 (revisited global graphs functionality).  [https://lnkd.tech/index.php/help/editor-plugin-help-global-graphs](https://lnkd.tech/index.php/help/editor-plugin-help-global-graphs)
  -  The "Dereference" quick fix should work for [https://stuchalk.github.io/scidata/ontology/chemtwin.owl](https://stuchalk.github.io/scidata/ontology/chemtwin.owl)
- *.OWL files are still not supported as editable files. However The "Global Graph" functionality is capable of transforming OWL file to TTL. During the download. 

What is missing: 
-  #54 
-  #46

**Comment by stuchalk** [comment]
Thanks for the version update and the note above.  I will try it out!
